godef.go

```diff
@@ -225,7 +227,24 @@ func (o orderedObjects) Swap(i, j int)      { o[i], o[j] = o[j], o[i] }
 func done(obj *ast.Object, typ types.Type) {
 	defer os.Exit(0)
 	pos := types.FileSet.Position(types.DeclPos(obj))
-	fmt.Printf("%v\n", pos)
+	if *jsonFlag {
+		var p  = struct {
+            Filename string `json:"filename,omitempty"`
+            Line int `json:"line,omitempty"`
+            Column int `json:"column,omitempty"`
+        }{
+            Filename: pos.Filename,
+            Line: pos.Line,
+            Column: pos.Column,
+        }
+		jsonStr, err := json.Marshal(p)
+		if err != nil {
```

Comment:
This should never happen. Better to call

    fail("JSON marshal error: %v", err)

if it does.